aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -241,14 +255,23 @@ module aptos_framework::code {
                             error::invalid_argument(EDEP_ARBITRARY_NOT_SAME_ADDRESS)
                         )
                     };
-                    found = true;
+                    // Add allowed deps
+                    let k = 0;
+                    let r = vector::length(&dep_pack.modules);
+                    while (k < r) {
```

Comment:
How much gas does this add?

Reply:
Should we consider Moving this to Rust?

Reply:
I guess marginally compared to overall cost. We had to load the registry anyway, so this data is there. Now we build a list of perhaps a few dozen entries. (We do not add all framework modules.) Whereas the code which is for example, touched in detail by the bytecode verifier, is a a few k large.